quickjs/src/main/jni/JsMethodProxy.cpp

```diff
@@ -37,6 +37,62 @@ std::string getName(JNIEnv *env, jclass javaClass) {
 JsMethodProxy::ArgumentLoader
 getJavaToJsConverter(const Context* context, jclass type, bool boxed) {
   const auto typeName = getName(context->env, type);
+  if (!typeName.empty() && typeName[0] == '[') {
+    // type is an array.
+    const jmethodID method = context->env->GetMethodID(context->env->GetObjectClass(type),
+                                                       "getComponentType",
+                                                       "()Ljava/lang/Class;");
+    auto elementType = static_cast<jclass>(context->env->CallObjectMethod(type, method));
+    const auto elementTypeName = getName(context->env, elementType);
+    if (elementTypeName == "double") {
+      return [](const Context* c, jvalue v) {
+        if (!v.l) return JS_NULL;
+        JSValue result = JS_NewArray(c->jsContext);
+        auto elements = c->env->GetDoubleArrayElements(static_cast<jdoubleArray>(v.l), nullptr);
+        for (jsize i = 0, e = c->env->GetArrayLength(static_cast<jarray>(v.l)); i < e; i++) {
+          JS_SetPropertyUint32(c->jsContext, result, i, JS_NewFloat64(c->jsContext, elements[i]));
+        }
+        c->env->ReleaseDoubleArrayElements(static_cast<jdoubleArray>(v.l), elements, JNI_ABORT);
+        return result;
+      };
+    } else if (elementTypeName == "int") {
+      return [](const Context* c, jvalue v) {
+        if (!v.l) return JS_NULL;
+        JSValue result = JS_NewArray(c->jsContext);
+        auto elements = c->env->GetIntArrayElements(static_cast<jintArray >(v.l), nullptr);
+        for (jsize i = 0, e = c->env->GetArrayLength(static_cast<jarray>(v.l)); i < e; i++) {
+          JS_SetPropertyUint32(c->jsContext, result, i, JS_NewInt32(c->jsContext, elements[i]));
+        }
+        c->env->ReleaseIntArrayElements(static_cast<jintArray>(v.l), elements, JNI_ABORT);
+        return result;
+      };
+    } else if (elementTypeName == "boolean") {
```

Comment:
Feels like there's missing test cases for these new types

Reply:
QuickJsGetTest.proxyFunctionObject()